tests/suites/test_suite_x509parse.data

```diff
@@ -2680,6 +2680,65 @@ mbedtls_x509_csr_parse:"308201193081bf0201003034310b3009060355040613024e4c311130
 X509 CSR ASN.1 (invalid version overflow)
 mbedtls_x509_csr_parse:"3008300602047fffffff":"":MBEDTLS_ERR_X509_UNKNOWN_VERSION
 
+# Used test_csr_v3_all.csr as a base for malforming CSR extenstions/attributes
+# Please see makefile for data_files to check malformation details (test_csr_v3_all_malformed_xxx.csr files)
+X509 CSR ASN.1 (attributes: invalid sequence tag)
+mbedtls_x509_csr_parse:"308201B13082011A020100300F310D300B06035504030C046574636430819F300D06092A864886F70D010101050003818D0030818902818100C73892C5C3F47010086BF81335ECF3011C8A250F9582361EAA1E9612551AACF87B75330B7057339CD995F14D4C4437C8BEC4A03FE4643CD3F3C902433DC37C5B8E79AA0984AE8213370ABF8569BD2A35626CB1FEE4ACE042B2417C7BE49D27149B2C076B8E433D3F6C763A469A1C700D259DBC08C740FCD8023762FF7DEBFC2D0203010001A062406006092A864886F70D01090E31533051300B0603551D0F040403020102302F0603551D1104283026A02406082B06010505070804A018301606072B060104011103040B3132338081008180333231301106096086480186F8420101040403020780300D06092A864886F70D01010B05000381810093D6BEE3D49597074924A061BDF0D0D08EA9A81F94825B20EE3BE8008808430639C11CB8636C13E023EEDF31B247142582A3C7FC92F766688BF4DA185CCACC6C9E12CBCDB2DA2C96DF542AF2D6A867A8C97E6BAC994454EEE40F38AD60FC8A36AF2BA6B86465B082B569AFDAA9E606D71B37E3E174517DE91B17FE9C87905EE3":"":MBEDTLS_ERR_X509_INVALID_EXTENSIONS+MBEDTLS_ERR_ASN1_UNEXPECTED_TAG
```

Comment:
I think that instead of duplicating the test data here, it would be better to change an existing test function (that already takes a file parameter) and add a possibility to make it fail. Look at `x509_parse_san`, or `x509_parse_san` from https://github.com/Mbed-TLS/mbedtls/pull/6900/commits/20d5bff61730b5a70389a4bd7da6f920601b0d25.
This data duplication is a blocker now, since most of the .csr files aren't even used now.

Reply:
I was considering changing the tests function to use file as input, but for consistency with other `X509 CSR ASN.1...` tests that takes input directly I decided to not make exception for the new tests.
I disagree that the generated csr files are useless. They are important as they inform how the malformed csr was created. At the beginning of test cases there is a comment informing about this.

Reply:
Yes, but now the data is manually duplicated in many places. Let's say that the basic certificate that served as the base for malforming needs to be updated (just created anew, to have a later expiry date) - all of these test cases would need to be recreated by hand, and it's not even that the same data would need to be copied in multiple places - each of the malformed files would need to be transformed from DER to PEM and then copied! That's a lot of manual work...

Reply:
> each of the malformed files would need to be transformed from DER to PEM and then copied!

No this is DER format. But generally yes after regeneration data need to be copied from `malformed.csr.der` to test data file. 

What do you suggest. Creating new function or adding parameter (determine if input is provided directly or by file) to `mbedtls_x509_csr_parse`?

Reply:
How about changing `mbedtls_x509_csr_info` the same way I modified https://github.com/Mbed-TLS/mbedtls/commit/20d5bff61730b5a70389a4bd7da6f920601b0d25#diff-fdcf8ae37b642bc33af08b33f6dbe62c1592bc976df8a276af76c030077d0bb3L429 ?